Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/WatchManager.java

```diff
@@ -254,8 +416,10 @@ public void reorder(List<Pin> pins) {
             List<Pin> l = new ArrayList<>();
 
             for (Pin p : pins) {
-                if (p.watching)
+                // User may have unpressed the watch thread button but there may still be flags
+                if (p.watching || PinType.hasFlags(p.pinType)) {
```

Comment:
Do not forget to update pin.pinType when unpinning/stopping to download a thread.

Reply:
Apparently not needed anymore.